packages/atauth/src/auth_cli.c

```diff
@@ -52,16 +52,33 @@ int main(int argc, char *argv[]) {
   const size_t base64_encoded_rsa2048_ciphertext_size = atchops_base64_encoded_size(rsa_2048_ciphertext_size);
   unsigned char *encrypted_apkam_symmetric_key_base64[base64_encoded_rsa2048_ciphertext_size];
 
-  // init buffers for IV's that will be used to decrypt keys received from server
-  unsigned char enc_privkey_iv[ATCHOPS_IV_BUFFER_SIZE];
-  unsigned char self_enc_key_iv[ATCHOPS_IV_BUFFER_SIZE];
+  // init buffer to hold encrypted default encryption private key received from server
+  const size_t rsa_2048_privkey_base64_len = atchops_base64_encoded_size(ATAUTH_RSA_2048_PRIVKEY_BYTES);
+  const size_t aes256_encrypted_rsa_privkey_size = atchops_aes_ctr_ciphertext_size(
+      rsa_2048_privkey_base64_len); // size for an AES256 encrypted RSA2048 privkey in bytes
+  const size_t aes256_encrypted_rsa_privkey_base64_size =
+      atchops_base64_encoded_size(aes256_encrypted_rsa_privkey_size);
+  char encrypted_default_encryption_private_key[aes256_encrypted_rsa_privkey_base64_size];
+
+  // init buffer to hold encrypted default self encryption key received from server
+  const size_t aes256_encrypted_aes_key_base64_size =
+      atchops_base64_encoded_size(atchops_aes_ctr_ciphertext_size(ATCHOPS_AES_256));
+  char encrypted_default_self_encryption_key[aes256_encrypted_aes_key_base64_size];
+
+  // init buffers to hold IVs for encrypted self enc key and enc private key received from server
+  size_t base64_encoded_iv_size = atchops_base64_encoded_size(ATCHOPS_IV_BUFFER_SIZE);
+  unsigned char enc_privkey_iv[base64_encoded_iv_size];  // base64 encoded iv
+  unsigned char self_enc_key_iv[base64_encoded_iv_size]; // base64 encoded iv
 
   memset(apkam_symmetric_key_bytes, 0, aes256_key_unsigned_char_bytes_size);
   memset(apkam_symmetric_key_base64, 0, aes256_key_unsigned_char_base64_size);
-  memset(encrypted_apkam_symmetric_key_bytes, 0, sizeof(unsigned char) * rsa_2048_ciphertext_size);
-  memset(encrypted_apkam_symmetric_key_base64, 0, sizeof(unsigned char) * base64_encoded_rsa2048_ciphertext_size);
-  memset(enc_privkey_iv, 0, sizeof(unsigned char) * ATCHOPS_IV_BUFFER_SIZE);
-  memset(self_enc_key_iv, 0, sizeof(unsigned char) * ATCHOPS_IV_BUFFER_SIZE);
+  memset(encrypted_apkam_symmetric_key_bytes, 0, rsa_2048_ciphertext_size);
+  memset(encrypted_apkam_symmetric_key_base64, 0, base64_encoded_rsa2048_ciphertext_size);
+  memset(encrypted_default_encryption_private_key, 0, aes256_encrypted_rsa_privkey_base64_size);
+  memset(encrypted_default_self_encryption_key, 0, aes256_encrypted_aes_key_base64_size);
+  memset(enc_privkey_iv, 0, base64_encoded_iv_size);
+  memset(self_enc_key_iv, 0, base64_encoded_iv_size);
+  memset(enrollment_id, 0, sizeof(enrollment_id));
```

Comment:
When you're doing memset, you should use `sizeof()` in the size argument.

For example:

```bash
// current:
memset(encrypted_apkam_symmetric_key_base64, 0, base64_encoded_rsa2048_ciphertext_size);
// should be:
memset(encrypted_apkam_symmetric_key_base64, 0, sizeof(char) * base64_encoded_rsa2048_ciphertext_size);
```